### PR TITLE
kinesis connection retry mechanism added into the current consumer

### PR DESCRIPTION
**Pull request by hamed-hatami**
# Description

# Target

- [ ] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [ ] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [x] I checked that each commit in the pull request has a meaningful subject line and body.

- [x] I have run `mvn clean install -DskipTests` locally and I have committed all auto-generated changes

### CODE REVIEW

File: components/camel-aws/camel-aws2-kinesis/src/main/java/org/apache/camel/component/aws2/kinesis/Kinesis2Endpoint.java [hunk 1]
```diff
@@ -99,12 +101,20 @@ public Producer createProducer() throws Exception {
...
-        final Kinesis2Consumer consumer = new Kinesis2Consumer(this, processor);
+        var kinesisConnection = KinesisConnection.getInstance();
+        final Kinesis2Consumer consumer = new Kinesis2Consumer(this, processor, kinesisConnection);
         consumer.setSchedulerProperties(getSchedulerProperties());
+        startHealthChecks(kinesisConnection);
         configureConsumer(consumer);
         return consumer;
     }
 
+    private void startHealthChecks(KinesisConnection kinesisConnection) {
```
Comment (oscerd) [inline components/camel-aws/camel-aws2-kinesis/src/main/java/org/apache/camel/component/aws2/kinesis/Kinesis2Endpoint.java:112]: For this you could use the executor service we have in Camel, like we did here: 

https://github.com/apache/camel/blob/main/components/camel-aws/camel-aws2-s3/src/main/java/org/apache/camel/component/aws2/s3/stream/AWS2S3StreamUploadProducer.java#L83

**Review by oscerd — CHANGES_REQUESTED** [review]
Just a note.

File: components/camel-aws/camel-aws2-kinesis/src/main/java/org/apache/camel/component/aws2/kinesis/Kinesis2Endpoint.java (hunk 1 above)
Comment (hamed-hatami) [reply components/camel-aws/camel-aws2-kinesis/src/main/java/org/apache/camel/component/aws2/kinesis/Kinesis2Endpoint.java:114]: What a great idea, I've changed the code according to your suggestion

Comment (hamed-hatami) [reply components/camel-aws/camel-aws2-kinesis/src/main/java/org/apache/camel/component/aws2/kinesis/Kinesis2Endpoint.java]: @oscerd  it's ready to go, I suppose

Comment (hamed-hatami) [reply components/camel-aws/camel-aws2-kinesis/src/main/java/org/apache/camel/component/aws2/kinesis/Kinesis2Endpoint.java]: @Andrea Cosentino Please retrigger the pipeline one more time

**Review by oscerd — APPROVED** [review]

**Comment by davsclaus** [comment]
Sorry I think this PR needed some attention but I did not have time today.

This kind of health check is not standard way in Camel and I would like to revert this and discuss more how to do this in a better way.

**Comment by hamed-hatami** [comment]
No problem if you wanna revert it but I see no specific way in camel to do fault tolerance, I wanted to use MP fault tolerance such as retry mechanism, I found nothing at that level, maybe in Quarkus level you can do it much more easier

**Comment by hamed-hatami** [comment]
Even if there was a way to check the healthiness of Kinesis but still it's tricky not to check the specific stream channel not to be chocked and working properly

**Comment by hamed-hatami** [comment]
BTW, something regarding the connection in previous implementation before this PR, it gets a new connection almost for every either getClient() or getAsyncClient() whereas within new imple , it's a singleton connection which takes care of single connection till it's down

**Pull request merged**